schema-engine/sql-schema-describer/src/postgres/indexes_query.sql

```diff
@@ -42,5 +44,6 @@ FROM
     INNER JOIN pg_am AS indexaccess ON indexaccess.oid = indexinfo.relam
     LEFT JOIN pg_opclass AS opclass -- left join because crdb has no opclasses
         ON opclass.oid = rawindex.indclass
+    LEFT JOIN pg_constraint pc ON rawindex.indexrelid = pc.conindid AND pc.contype <> 'f'
```

Comment:
Do we have anywhere / could you point me in the right direction for where I can read up about these?
There just being an exclusion check for `f` seems very arcane and magical to me 🤔

Reply:
Hehe, yes it is arcane magic. So for foreign keys we get an index constraint to the table. If we don't filter them out, we get some indices twice :P

Reply:
ahh, so f == is the foreign key constraint type
I'm guessing that's what `.contype` means

Reply:
I think you are looking for this @Druue : https://www.postgresql.org/docs/current/catalog-pg-constraint.html

Reply:
Yeah, and we don't care about those in the index query.